Packages/com.unity.inputsystem/InputSystem/Plugins/UI/UIActionInputModule.cs

```diff
@@ -199,6 +326,29 @@ private void UnhookActions()
                 cancelAction.performed -= m_ActionCallback;
         }
 
+        private void SwapProperty(ref InputActionProperty oldProperty, InputActionProperty newProperty)
+        {
+            if (oldProperty != null)
+            {
+                if (m_ActionsEnabled)
+                    oldProperty.action.Disable();
+
+                if (m_ActionsHooked)
+                    oldProperty.action.performed -= m_ActionCallback;
+            }
+
+            oldProperty = newProperty;
+
+            if (oldProperty != null)
+            {
+                if (m_ActionsHooked)
+                    oldProperty.action.performed += m_ActionCallback;
+
+                if (m_ActionsEnabled)
+                    oldProperty.action.Enable();
```

Comment:
Enabling and disabling actions should not be under the control of the UI input module. The input module should solely be a consumer.